cmd/run_java.go

```diff
@@ -49,14 +44,40 @@ dependencies {
 application {
     mainClass = 'io.temporal.sdkfeatures.Main'
 }`
-	if err := os.WriteFile(filepath.Join(tempDir, "build.gradle"), []byte(buildGradle), 0644); err != nil {
+	if err := os.WriteFile(filepath.Join(p.config.Dir, "build.gradle"), []byte(buildGradle), 0644); err != nil {
 		return fmt.Errorf("failed writing build.gradle: %w", err)
 	}
-	settingsGradle := fmt.Sprintf("rootProject.name = '%v'", filepath.Base(tempDir))
-	if err := os.WriteFile(filepath.Join(tempDir, "settings.gradle"), []byte(settingsGradle), 0644); err != nil {
+	settingsGradle := fmt.Sprintf("rootProject.name = '%v'", filepath.Base(p.config.Dir))
+	if err := os.WriteFile(filepath.Join(p.config.Dir, "settings.gradle"), []byte(settingsGradle), 0644); err != nil {
 		return fmt.Errorf("failed writing settings.gradle: %w", err)
 	}
 
+	// Build if wanted
+	if build {
+		// This is really only to prime the system-level caches. The build won't be
+		// used by run.
+		return runGradle(ctx, p.log, p.config.Dir, []string{"--no-daemon", "--include-build", "../", "build"})
+	}
+	return nil
+}
+
+// RunJavaExternal runs the Java run in an external process. This expects the
+// server to already be started.
+func (r *Runner) RunJavaExternal(ctx context.Context, run *cmd.Run) error {
+	// Create temp dir if needed and prepare the project
+	if r.config.Dir == "" {
+		var err error
+		if r.config.Dir, err = os.MkdirTemp(r.rootDir, "sdk-features-java-test-"); err != nil {
+			return fmt.Errorf("failed creating temp dir: %w", err)
+		}
+		r.createdTempDir = &r.config.Dir
+
+		// Prepare the project but don't build since it'll happen when we run later
+		if err := NewPreparer(r.config.PrepareConfig).PrepareJavaExternal(ctx, false); err != nil {
```

Comment:
Why not build here? Does it slow things down? Just curious..

Reply:
Because we don't use the build. The "build" is just to prime the system caches. There is no reason, due to how quick Java is, to rework this code to use the built JAR instead of just using `gradle run`.